src/mon/PGMap.cc

```diff
@@ -2363,6 +2363,26 @@ void PGMap::dump_pool_stats_and_io_rate(int64_t poolid, const OSDMap &osd_map,
   }
 }
 
+// Get crush parentage for an osd (skip root)
+set<std::string> PGMap::osd_parentage(const OSDMap& osdmap, int id) const
+{
+  set<std::string> reporters_by_subtree;
+  // XXX: We really want whatever has OSDs
+  auto reporter_subtree_level = g_conf().get_val<string>("mon_osd_reporter_subtree_level");
+
+  auto loc = osdmap.crush->get_full_location(id);
+  for (map<string,string>::iterator p = loc.begin(); p != loc.end(); ++p) {
+    if (p->first == "root") continue;
+    if (p->first != reporter_subtree_level) {
+      reporters_by_subtree.insert(p->second);
+    }
+  }
+  for (auto s: reporters_by_subtree)
+	  dout(0) << __func__ << " " << s << dendl;
+  dout(0) << __func__ << " set " << reporters_by_subtree << dendl;
```

Comment:
can we use a higher debug level for this message? as to my understanding, this is not strictly an error.

Reply:
mistake

Reply:
Code added for development testing, removed.